Intersect.Client/Entities/Player.cs

```diff
@@ -577,28 +581,43 @@ public void TrySellItem(int index)
                 if (foundItem > -1 && Globals.GameShop.BuyingWhitelist ||
                     foundItem == -1 && !Globals.GameShop.BuyingWhitelist)
                 {
-                    if (Inventory[index].Quantity > 1)
+                    if (quantity > 1)
                     {
-                        var iBox = new InputBox(
-                            Strings.Shop.sellitem,
-                            Strings.Shop.sellitemprompt.ToString(ItemBase.Get(Inventory[index].ItemId).Name), true,
-                            InputBox.InputType.NumericSliderInput, SellItemInputBoxOkay, null, index, Inventory[index].Quantity
+                        InputBox.Open(
+                            title: Strings.Shop.sellitem,
+                            prompt: Strings.Shop.sellitemprompt.ToString(sellingItem.Name),
+                            modal: true,
+                            inputType: InputBox.InputType.NumericSliderInput,
+                            onSuccess: SellItemInputBoxOkay,
+                            onCancel: null,
+                            userData: index,
+                            quantity: quantity,
+                            maxQuantity: quantity
                         );
                     }
                     else
                     {
-                        var iBox = new InputBox(
-                            Strings.Shop.sellitem,
-                            Strings.Shop.sellprompt.ToString(ItemBase.Get(Inventory[index].ItemId).Name), true,
-                            InputBox.InputType.YesNo, SellInputBoxOkay, null, index
+                        InputBox.Open(
+                            title: Strings.Shop.sellitem,
+                            prompt: Strings.Shop.sellprompt.ToString(sellingItem.Name),
+                            modal: true,
+                            inputType: InputBox.InputType.YesNo,
+                            onSuccess: SellInputBoxOkay,
+                            onCancel: null,
+                            userData: index
                         );
                     }
                 }
                 else
                 {
-                    var iBox = new InputBox(
-                        Strings.Shop.sellitem, Strings.Shop.cannotsell, true, InputBox.InputType.OkayOnly, null, null,
-                        -1
+                    InputBox.Open(
+                        title: Strings.Shop.sellitem,
+                        prompt: Strings.Shop.cannotsell,
+                        modal: true,
+                        inputType: InputBox.InputType.OkayOnly,
+                        onSuccess: null,
+                        onCancel: null,
+                        userData: -1
```

Comment:
Instead of 3 InputBox.Opens, can we use the if statements to save all of the things that can change (prompt, input type, success, cancel, user data, quantity, max quantity) to variables and then only call InputBox.Open once?